<div class="holy-grail-flexbox">
    <header class="header">Async With Blok Method: turbo_frame_tag_async "tag", data1, data2 do |async_tag|</header>

    <section class="left-sidebar">
      <p> Default `loading`, `failure` msgs </p>

      <%= turbo_frame_tag_async "widget2", data1, data3 %>
    </section>

    <main class="main-content">
      <p> Custom `loading`, `success` msgs </p>

      <%= turbo_frame_tag_async "widget1", data1, data2 do |d1, d2| %>
        <div class="success">
          Data 1: <%= d1 %>
          Data 2: <%= d2 %>
        </div>
      <% end.on_loading do %>
        <div class="loading">
          <span>Loading Widget...</span>
        </div>
      <% end.on_failure do |errors| %>
        <div class="error">
          Something went wrong: <%= errors.to_s %>
        </div>
      <% end %>
    </main>

    <aside class="right-sidebar">
      <p> Custom `loading`, `failure` msgs </p>

      <%= turbo_frame_tag_async "widget3", data2, data3 do |d2, d3| %>
        You should not see this text. Block should fail!
      <% end.on_loading do %>
        <div class="loading">
          <span>Loading Widget with custom error...</span>
        </div>
      <% end.on_failure do |errors| %>
        <div class="error">
          Something went wrong: <%= errors.to_s %>
        </div>
      <% end %>
    </aside>

    <footer class="footer"></footer>
</div>
